plugins/parcelport/verbs/connection_handler.hpp

```diff
@@ -0,0 +1,92 @@
+//  Copyright (c) 2015 John Biddiscombe
+//
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef HPX_PARCELSET_POLICIES_CONNECTION_HANDLER_HPP
+#define HPX_PARCELSET_POLICIES_CONNECTION_HANDLER_HPP
+
+#include <hpx/config/warnings_prefix.hpp>
+#include <hpx/runtime/parcelset/locality.hpp>
+#include <hpx/runtime/parcelset/parcelport_impl.hpp>
+//
+#include <plugins/parcelport/verbs/rdma/rdma_logging.hpp>
+#include <plugins/parcelport/verbs/rdma/rdma_error.hpp>
+#include <plugins/parcelport/verbs/rdma/verbs_device.hpp>
+//
+#include <sstream>
+#include <type_traits>
+
+namespace hpx {
+namespace parcelset {
+namespace policies {
+namespace verbs
+{
+    class HPX_EXPORT parcelport;
+}}
+
+    template <>
+    struct connection_handler_traits<policies::verbs::parcelport>
+    {
+        typedef policies::verbs::sender_connection      connection_type;
+        typedef HPX_PARCELPORT_VERBS_HAVE_BOOTSTRAPPING send_early_parcel;
+        typedef std::true_type                          do_background_work;
+        typedef std::true_type                          send_immediate_parcels;
+
+        static const char * type()
+        {
+            return "verbs";
+        }
+
+        static const char * pool_name()
+        {
+            return "parcel-pool-verbs";
+        }
+
+        static const char * pool_name_postfix()
+        {
+            return "-verbs";
+        }
+    };
+
+namespace policies {
+namespace verbs
+{
+        uint32_t get_verbs_device_address(const char *devicename, const char *iface,
+            char *hostname)
+        {
+          FUNC_START_DEBUG_MSG
+          // Find the address of the I/O link device.
+          verbs_device_ptr linkDevice;
+          try {
+            linkDevice = verbs_device_ptr(new verbs_device(devicename, iface));
```

Comment:
good question. I will check if I reuse the device and make it a static instantiation if I don't need a smart_pointer.